.github/workflows/run-mill-action.yml

```diff
@@ -53,10 +53,10 @@ jobs:
 
       - uses: coursier/cache-action@v6
 
-      - uses: actions/setup-java@v4
+      - uses: coursier/setup-action@v1
         with:
-          java-version: ${{ inputs.java-version }}
-          distribution: temurin
+          apps: scalafmt
+          jvm: temurin:${{ inputs.java-version }}
```

Comment:
I'd like to avoid adding a dependency on a global `scalafmt` binary. Adding it in CI will necessitate people installing it locally to match the CI behavior, doubles the number of things that need to be installed (previously it was only 1: the JVM).

We already have Scalafmt on the Mill classpath, so instead of installing it globally we could define a top-level command in `build.mill` that uses Scalafmt programmatically (either inprocess or subprocess) on the files in `T.workspace`. That would achieve the same thing but maintain Mill's minimal unmanaged dependency footprint

Reply:
Alternatively, if we want to bypass mill entirely (e.g. for performance), we could have a `ci/scalafmt.sh` script to download the ScalaFmt graal native executable appropriate for your OS, cache it somewhere, and run that

Reply:
That's what I did in the last push